keps/sig-api-machinery/20190226-network-proxy.md

```diff
@@ -332,11 +332,12 @@ Given that we use a ‘master’ NetworkContext.
 - **ImagePolicyWebhook**
 The image policy webhook uses a kube config file to determine destination.
 Given that we use a ‘master’ NetworkContext.
-- **KMS GRPC Service**
-KMS connects with an ‘endpoint’ (not the resource) via gRPC.
+- ~~**KMS GRPC Service**~~
+~~KMS connects with an ‘endpoint’ (not the resource) via gRPC.
 The service at the endpoint provides the secret information for use in encryption.
 This is not a user space configurable system.
-Given that we use a ‘master’ NetworkContext.
+Given that we use a ‘master’ NetworkContext.~~
```

Comment:
Note that KMS currently only supports a [dialer][] using unix domain sockets (UDS), which is for communication between processes running on same host operating system. Thus, it is not a part of the egress traffic.

[dialer]:https://github.com/kubernetes/kubernetes/blob/e8bc121341807f9e33a076f6725b1b1a18d75ba0/staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service.go#L74

We may consider adding egressSelector support if KMS supports other protocol in the future.